core/bench_contract_test.go

```diff
@@ -0,0 +1,173 @@
+// (c) 2020-2021, Ava Labs, Inc.
+//
+// This file is a derived work, based on the go-ethereum library whose original
+// notices appear below.
+//
+// It is distributed under a license compatible with the licensing terms of the
+// original code from which it is derived.
+//
+// Much love to the original authors for their work.
+// **********
+// Copyright 2014 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package core
+
+import (
+	_ "embed"
+	"math/big"
+	"testing"
+
+	"github.com/ava-labs/subnet-evm/core/state"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/core/vm"
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ava-labs/subnet-evm/precompile/contract"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	//go:embed TrieStressTest.bin
+	stressBinStr string
+	//go:embed TrieStressTest.abi
+	stressABIStr string
+)
+
+func BenchmarkTrie(t *testing.B) {
+	benchInsertChain(t, true, stressTestTrieDb(t, 100, 6, 50, 1202102))
```

Comment:
what is the target we want to benchmark here? This currently inserts `N` number of blocks but these seems static (numContracts, callsPerBlock, elements). Should these be depending on `N`?

Reply:
or do we just want to see how total writes impact the benchmark (N * callsPerBlock * elements?)

Reply:
We want to deploy N smart contract instances (100), Call 6 different contracts per block (and measure their performance). Each call will append 50 new elements (this logic is part of the solidity code). The last argument is how much gas should be used per each call.

Reply:
So we don't want these to be anyhow dynamic with benchmark value (`N`)? 
I'm a bit confused at "We want to deploy N smart contract instances (100)". This won't be same with `t.N` which is the benchmark iteration value. I.e BenchmarkTrie-8 BenchmarkTrie-16 where N=8, 16 etc.

Reply:
@ceyonur not really.  The idea is to make changes to the underlying library and see how the benchmark changes with commit changes.